### DISCUSSION TITLE
IPED 4.1.3 Performance

### DISCUSSION THREAD

**Discussion by gisms** [sepinf-inc/IPED#1785] [Q&A] [answered]
We have a ufdr file to index, 130GB.
In version 4.1.3 processing is taking days, with overhead in audio transcription.
Already in version 4.1.1 it ends in hours.

One thing we are not able to increase in version 4.1.3 is processing ram. We tried to increase -Xms, but it is limited to 8GB, we believe this is impacting indexing time. We have workstation to process with 64GB.

Could you help to optimize this performance in 4.1.3?

**Comment by lfcnassif** [comment, marked as answer]
> One thing we are not able to increase in version 4.1.3 is processing ram. We tried to increase -Xms, but it is limited to 8GB, we believe this is impacting indexing time. We have workstation to process with 64GB.

Try to specify -Xmx instead of -Xms


> In version 4.1.3 processing is taking days, with overhead in audio transcription.

Audio transcription is a very heavy task and it may take days depending on your input if you don't have a good GPU to help.

**Reply by gisms** [reply]
I am indexing rigth now with 4.1.3 and thats how is my gpu usage:

![image](https://github.com/sepinf-inc/IPED/assets/66680566/11244152-2e66-4c52-a01a-58dc0ef11147)

i thinking is not using it.
is there a manual to configure audiotranscipt to use it?

**Reply by lfcnassif** [reply]
Seems it's not, GPU memory is not being used.  Default transcription implementation (Vosk) doesn't use GPU. You have to change to Wav2vec2 algorithm, which has a much better accuracy, take a look at the wiki manual.

And to use Wav2Vec2 with your GPU, there are some tips here: #1362

**Reply by gisms** [reply]
ok, thank u, i will do some tests.

**Comment by gisms** [comment]
"Audio transcription is a very heavy task and it may take days depending on your input if you don't have a good GPU to help."
Yes but we used it in previous versions and it was not a problem.
I am testing 4.1.3 with xmx and i will return to a feedback.

**Reply by lfcnassif** [reply]
Are you processing the same evidence with the exact same options? As I said, transcription time depends on the input.

**Reply by gisms** [reply]
yes, it was the same. Increasing the xmx, we return to a normal state of time to index.
